lib/galaxy/managers/users.py

```diff
@@ -227,9 +227,10 @@ def purge(self, user, flush=True):
             if self.app.config.redact_email_during_deletion:
                 role.name = role.name.replace(user.email, email_hash)
                 role.description = role.description.replace(user.email, email_hash)
-            private_role.name = email_hash
-            private_role.description = f"Private Role for {email_hash}"
-            self.session().add(private_role)
+            self.session().add(role)
```

Comment:
I've added this as I think it was missing?

Reply:
It should be pulled in through the session relationship (at least on sqlalchemy 1.4), but it's good to add. Can you assign the session and reuse it ?

Reply:
I'll refactor that later, a lot of what's happening in that method is unnecessary

Reply:
Thanks!